atlp-rwanda/ecommerce-app-predators-bn
# #10 ft(Auto-action): password change prompt

## What does this PR do?

   This PR adds a password update feature to the app that prompts users to update their password after a certain time period 
   and records the time of the last password update.

## Description of Task Completed

   Created a field in the user/profile model to store the last password update time.
   Implemented a cron job to iterate through all users and check if the password expiration time has been reached.
   Implemented Node events to trigger the password update prompt when the expiration time is reached and prevent further operations until the password is updated.

## How should this be manually tested?

   Log in as a user and wait for the password expiration time to be reached.
   The app should prompt you to update your password and record the time of the last password update.

   **If the prompt is ignored, the app should prevent further operations until the password is updated.**

```
git clone https://github.com/atlp-rwanda/ecommerce-app-predators-bn.git
cd ecommerce-app-predators-bn
git checkout ft-auto-actions-password-prompot-10
npm i
npm run dev
```

## Any background context you want to provide?
  This feature enhances the security of the app by enforcing regular password updates and recording the last password 
   update time.

## What are the relevant pivotal tracker stories?

Trello [Link](https://trello.com/c/MdMdiS3i/10-auto-action-password-change-prompt) (#-10)

## Screenshots (if appropriate)
